### ISSUE TITLE
Music player problems

### ISSUE THREAD

**Issue by NicTanghe** [linkfrg/dotfiles#23]
Hello thanks for making such an awsome and neat example of an eww configuration.
I`ve been trying to adjust it to my personal system.

Im currently trying to adjust the musicplayer as I`m having some issues with it.

this is my error:

```yuck
error: Failed to turn `` into a value of type json-value
   ┌─ modules/music.yuck:19:72
   │
19 │                     (label :hexpand true :class "music-strtime" :text {mpris_positions[mpris.name].positionStr} :halign "start")
   │                                                                        ───────────────


 2024-06-09T13:33:45.223Z WARN  eww::script_var_handler > stderr of `mpris_positions`: jq: error: syntax error, unexpected ':', expecting '}' (Unix shell quoting issues?) at <top-level>, line 1:
 2024-06-09T13:33:45.223Z WARN  eww::script_var_handler > stderr of `mpris_positions`: {$player: {position: $position, positionStr: $positionStr}}
 2024-06-09T13:33:45.223Z WARN  eww::script_var_handler > stderr of `mpris_positions`: jq: error: May need parentheses around object key expression at <top-level>, line 1:
 2024-06-09T13:33:45.223Z WARN  eww::script_var_handler > stderr of `mpris_positions`: {$player: {position: $position, positionStr: $positionStr}}
 2024-06-09T13:33:45.223Z WARN  eww::script_var_handler > stderr of `mpris_positions`: jq: 2 compile errors
error: Failed to turn `` into a value of type json-value
   ┌─ modules/music.yuck:17:72
```

the player finds the song that is playing I can pause and scipt and everything but it doesn’t seem to know where in the track it is and it also never displays an album cover.

If you find the time to help me il make a 3d render of something u want or smth.

**Comment by martian0x80** [comment]
@NicTanghe what player do you use?
afaik, `playerctl metadata` doesn't return anything for `artUrl` on the lastest Spotify, or anything you play through a browser (visible as `plasma-browser-integration`). But, I managed to fix it for local players like Strawberry.
![image](https://github.com/linkfrg/dotfiles/assets/26498920/c40a5e67-1c88-42a6-b85d-5c119eae66e5)

**Comment by NicTanghe** [comment]
Hello i was using the gnome music player.
I have since switched to rocky linux.
where allot of things don`t work.

ps did ytou fix it after my post or ws something "wrong" with my donfig ?

**Comment by martian0x80** [comment]
@NicTanghe Hey, so I issued a patch on my fork of the repo [here](https://github.com/martian0x80/linkfrg-dotfiles-fix/commit/165f1bb4898959cbc9b6990f020388828d805dc0) (scroll down to the `mpris.sh`). This should fix the image/album cover not appearing for all local players like Strawberry and Gnome Music, I tested it on both and it works.

![image](https://github.com/linkfrg/dotfiles/assets/26498920/69b60331-1316-4825-996e-61b352067e80)

However, I can't reproduce your main issue tho. Will try to fix the weird timings on the play bar.
